helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -19,10 +19,277 @@
  * under the License.
  */
 
+import org.apache.helix.HelixException;
+import org.apache.helix.controller.dataproviders.ResourceControllerDataProvider;
+import org.apache.helix.model.ClusterConfig;
+import org.apache.helix.model.InstanceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.lang.Math.max;
+
 /**
- * A placeholder before we have the implementation.
- *
- * This class represents a potential allocation of the replication.
- * Note that AssignableNode is not thread safe.
+ * This class represents a possible allocation of the replication.
+ * Note that any usage updates to the AssignableNode are not thread safe.
  */
-public class AssignableNode { }
+public class AssignableNode {
+  private static final Logger _logger = LoggerFactory.getLogger(AssignableNode.class.getName());
+
+  // basic node information
+  private final String _instanceName;
+  private Set<String> _instanceTags;
+  private String _faultZone;
+  private Map<String, List<String>> _disabledPartitionsMap;
+  private Map<String, Integer> _maxCapacity;
```

Comment:
why not final field? Do we have capacity key model (class or enum)?

Reply:
Conceptually, these items are configurable. If we have long-live AssignableNode object (not the case now, but could be when we start optimizing), these fields will be updated.

Moreover, since all the set methods are private, there is no big difference if we mark it as final or not. Let me know if you have any real concern.

Reply:
The capacity key is string now.